olp-cpp-sdk-core/src/cache/ProtectedKeyList.cpp

```diff
@@ -41,10 +47,9 @@ bool ProtectedKeyList::Deserialize(KeyValueCache::ValueTypePtr value) {
   if (!value) {
     return false;
   }
-  std::stringbuf string_buf;
-  string_buf.pubsetbuf(reinterpret_cast<char*>(value->data()), value->size());
 
-  std::istream stream(&string_buf);
+  ReadBuffer buf(reinterpret_cast<char*>(value->data()), value->size());
+  std::istream stream(&buf);
```

Comment:
```suggestion
  std::istringstream stream({value->begin(), value->end()});
```
I think, `istringstream` fits good here and no need for cast and additional class

Reply:
maybe better do not copy protected list